roles/routeros_common/tasks/manage_users.yml

```diff
@@ -0,0 +1,102 @@
+---
+# Manage Mikrotik Users
+# Missing users are added with random password and ssh public key loaded.
+# Users present and in the delete_users list are delete.
+# For now, we leave extra users as is.
+
+    - name: setup expected users
+      set_fact:
+        expected_users: "{{ users_admin + users_user + ['admin'] }}"
+        delete_users: ['DeleteMe1', 'DeleteMe2']
+
+    - name: Run /user print command on Mikrotik
+      community.routeros.command:
+        commands: "/user print"
+      register: user_output
+      changed_when: false
+      check_mode: false
+
+    - name: Debug raw command output
+      debug:
+        var: user_output.stdout_lines[0]
+
+    - name: Extract usernames
+      set_fact:
+        mikrotik_users: >-
+          {{ user_output.stdout_lines[0] |
+             reject('search', ';;;') |
+             reject('search', ' # ') |
+             select('search', '^\s*\d*\s+E*\s*\S+') |
+             map('regex_replace', '^\s*\d*\s+E*\s*(\S+).*', '\1') |
+             list }}
+
+    - name: Display extracted usernames
+      debug:
+        var: mikrotik_users
+
+    - name: Find missing users
+      set_fact:
+        missing_users: >-
+          {{ expected_users | difference(mikrotik_users) }}
+
+    - name: Display missing users
+      debug:
+        var: missing_users
+
+    - name: Find extra users
+      set_fact:
+        extra_users: >-
```

Comment:
just want to verify my understanding of the desired behavior:
* ensure we have at least all the users in `expected_users` (which is the union of `users_admin` and `users_user`)
* remove any existing users in `deleted_users` (which is a hardcoded pair of users "DeleteMe1" and "DeleteMe2"?)
* display any other existing users not in `expected_users`, but leave them in place
* there's no distinction between users listed in `users_admin` vs `users_user` (at least for now)

is that correct?

Reply:
Basically yes.  I should change deleted users to something real or an empty list.  Will do that in the next commit.
I do treat the monitoring user specially.  I should use users_user to do that. Let's see if I can suss that out.

Reply:
Found and tested a fix for hardcoded monitoring user.  I now use users_user membership.